be/src/exprs/block_bloom_filter_impl.cc

```diff
@@ -83,17 +85,26 @@ Status BlockBloomFilter::init(const int log_space_bytes, uint32_t hash_seed) {
     return Status::OK();
 }
 
-Status BlockBloomFilter::init_from_directory(int log_space_bytes, const Slice& directory,
-                                             bool always_false, uint32_t hash_seed) {
+Status BlockBloomFilter::init_from_directory(int log_space_bytes,
+                                             butil::IOBufAsZeroCopyInputStream* data,
+                                             const size_t& data_size, bool always_false,
+                                             uint32_t hash_seed) {
```

Comment:
It may not be a good idea to change the interface like that which increases the number of arguments. `data` and `data_size` are related and users will use them at the same time here. Try to create some `struct` like `Slice`.

Reply:
we only provide users SQL interface. For developers, we can't continue to use Slice here because data blocks in IOBuf are not continuous. Besides, I think changing Slice to  IOBufAsZeroCopyInputStream is not different from adding an argument here.

Reply:
I'm not sure if it's necessary. `data_size ` here only means bloom filter's size so it may not be used anywhere. So do we really need to make this abstract? I have some doubts for this.

Reply:
Two points here:
1. The number of arguments a little more.
2. The following snippet (100-107) looks a little bit complex.

If you introduce another `struct` to wrap them, you can make the code more cleaner.

This comment aimed to enhance the readability other than functionality.